### PR TITLE
Remove deprecated cassandra flags

### PR DESCRIPTION
**Pull request by albertteoh**
Signed-off-by: albertteoh <albert.teoh@logz.io>

## Which problem is this PR solving?
- Relates to #2738

## Short description of the changes
- Removes deprecated cassandra flags

### CODE REVIEW

File: pkg/cassandra/config/config.go [hunk 1]
```diff
@@ -44,7 +44,6 @@ type Configuration struct {
 	Port                 int            `yaml:"port" mapstructure:"port"`
 	Authenticator        Authenticator  `yaml:"authenticator" mapstructure:",squash"`
 	DisableAutoDiscovery bool           `yaml:"disable_auto_discovery" mapstructure:"-"`
-	EnableDependenciesV2 bool           `yaml:"enable_dependencies_v2" mapstructure:"-"`
```
Comment (yurishkuro) [inline pkg/cassandra/config/config.go:47]: I expect there to be some code that actually reads this field, not seeing anything like that removed in the PR.

**Review by yurishkuro — COMMENTED** [review]
please add changelog entry

File: pkg/cassandra/config/config.go (hunk 1 above)
Comment (albertteoh) [reply pkg/cassandra/config/config.go:47]: Did a recursive case-insensitive search in my jaeger root project dir and couldn't find `EnableDependenciesV2` or `enable_dependencies_v2`.

Maybe something wrong with my search, can you see any references?

Comment (yurishkuro) [reply pkg/cassandra/config/config.go:47]: found it - the logic for that flag was already removed in #1364

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**